components/layout/construct.rs

```diff
@@ -1851,3 +1869,25 @@ fn control_chars_to_fragment(node: &InlineFragmentNodeInfo,
                                          restyle_damage,
                                          info)
 }
+
+/// Convenience methods for computed CSS values
+trait ComputedValueUtils {
+    /// Returns true if this node has non-zero padding or border.
+    fn has_padding_or_border(&self) -> bool;
+}
+
+impl ComputedValueUtils for ServoComputedValues {
+    fn has_padding_or_border(&self) -> bool {
+        let padding = self.get_padding();
+        let border = self.get_border();
+
+        padding.padding_top != LengthOrPercentage::Length(Au(0)) ||
+           padding.padding_right != LengthOrPercentage::Length(Au(0)) ||
+           padding.padding_bottom != LengthOrPercentage::Length(Au(0)) ||
+           padding.padding_left != LengthOrPercentage::Length(Au(0)) ||
```

Comment:
If that's going to be a thing we do, we should probably add an `is_zero()` method to `LengthOrPercentage` or something.